csi/src/freezefs.rs

```diff
@@ -0,0 +1,75 @@
+//! The files system freeze support using linux utility fsfreeze
+use crate::{
+    dev::{Device, DeviceError},
+    mount,
+};
+use snafu::{ResultExt, Snafu};
+use tokio::process::Command;
+use uuid::Uuid;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub(crate)")]
+pub enum FreezeFsError {
+    #[snafu(display("Cannot find volume: volume ID: {}", volid))]
+    VolumeNotFound { volid: String },
+    #[snafu(display("Invalid volume ID: {}, {}", volid, source))]
+    InvalidVolumeId {
+        source: uuid::parser::ParseError,
+        volid: String,
+    },
+    #[snafu(display("fsfreeze failed: volume ID: {}, {}", volid, error))]
+    FsfreezeFailed { volid: String, error: String },
+    #[snafu(display("Internal failure: volume ID:{}, {}", volid, source))]
+    InternalFailure { source: DeviceError, volid: String },
+    #[snafu(display("IO error: volume ID: {}, {}", volid, source))]
+    IOError {
+        source: std::io::Error,
+        volid: String,
+    },
+}
+
+const FSFREEZE: &str = "fsfreeze";
+
+async fn fsfreeze(
+    volume_id: &str,
+    freeze_op: &str,
```

Comment:
should this be an enum?

Reply:
Depends, I didn't see any point in doing that. The function is  internal to the freezefs, and I'd have to convert the enum to a string for the command line args, so elected to just pass the string itself.